src/java/fr/paris/lutece/portal/business/file/TemporaryFileHome.java

```diff
@@ -140,4 +140,22 @@ public static TemporaryFile findByPrimaryKey( int nKey )
     {
         return _dao.findByUser( user );
     }
+    
+    /**
+     * Delete the temporary files older than the given number of days
+     * @param days
+     */
+    public static void deleteFilesOlderThan( int days )
+    {
+        List<TemporaryFile> files = _dao.selectFilesOlderThan( days );
+        
+        for ( TemporaryFile file : files )
+        {
+            if ( file.getPhysicalFile( ) != null )
+            {
+                PhysicalFileHome.remove( file.getPhysicalFile( ).getIdPhysicalFile( ) );
+            }
+            _dao.delete( file.getIdFile( ) );
```

Comment:
These statements should be in a single transaction. If not, the physical file can be deleted when the tempory file isn't